### PR TITLE
fix: auto set vmid when creating host from Foreman API

### LINKED ISSUE #254: vmid not auto set when creating from Foreman API
**Describe the Bug**
Creating hosts through the web ui works correctly, however attempting to do it through the api returns the error `Backtrace for 'Action failed' error (ArgumentError): vmid is required for this operation`

**To Reproduce**
Attempt to create a host through the API with a host group that defines using a proxmox compute resource. Sample python used:
```
import requests, json
from requests.adapters import HTTPAdapter
from urllib3.util.retry import Retry
retry_strategy = Retry(
    total=5,
    status_forcelist=[429, 502, 503, 504],
    allowed_methods=["POST","GET"],
    backoff_factor=10
)
adapter = HTTPAdapter(max_retries=retry_strategy)
forehttp = requests.Session()
forehttp.auth = ("<foremanusername>","<foremanPAT>")
forehttp.mount('https://', adapter)
create_params={
    'host':{
        'name': <hostname>,
        'hostgroup_id': 28, # Host group that is set to use a proxmox compute profile
        'location_id': 5,
        'organization_id': 3,
        'build': True,
        'compute_attributes': {
            'start': 1,
           # 'vmid': '123'
        }
    }
}
r = forehttp.post("<foremanurl>/api/hosts",data=json.dumps(create_params),headers={'Content-Type': 'application/json'})
r.raise_for_status()
```

**Expected behavior**
As with the webui, the plugin should get the next vmid automatically. 

**Additional context**
If a valid VMID is specified manually (by uncommenting the `'vmid': 123` under `compute_attributes`) it works perfectly, but that is not easily workable for this script. Additionally passing any string to `vmid` that isn't a valid proxmox vmid results in the host being created in foreman, but silently failing to do anything in proxmox. Subsequent efforts to edit the VM  in foreman will return the error 
```
'<hostname>' not found on '<proxmox cluster>(Proxmox)' '<hostname>' could be deleted or '<proxmox cluster> (Proxmox)' is not responding.
```

**Versions**
Foreman: 3.6.1
rubygem-fog-proxmox.noarch: 0.15.0-1.el8
rubygem-foreman_fog_proxmox.noarch: 0.14.2-1.fm3_7.el8

### PR DESCRIPTION
**Pull request by Manisha15**
fixes #254

### CODE REVIEW

File: app/models/foreman_fog_proxmox/proxmox_vm_commands.rb [hunk 1]
```diff
@@ -31,11 +31,13 @@ def start_on_boot(vm, args)
       vm
     end
 
+    # rubocop:disable Metrics/AbcSize
     def create_vm(args = {})
       vmid = args[:vmid].to_i
       type = args[:type]
       node = client.nodes.get(args[:node_id])
       vmid = node.servers.next_id.to_i if vmid < 1
+      args[:vmid] = vmid.to_s if args[:vmid].nil?
```
Comment (sbernhard) [inline app/models/foreman_fog_proxmox/proxmox_vm_commands.rb:40]: Can you please explain? This looks strange - as in line 35 the vmid is set to `args[:vmid].to_i`.

to reduce the AbcSize, we may put everything to get/calculate the vmid in a separte method?

Comment (Manisha15) [reply app/models/foreman_fog_proxmox/proxmox_vm_commands.rb:40]: In line 35, we set vmid to args[:vmid] for clone_from_image part which is on line 46. But for API, args[:vmid] can be nil that's why it needs to be assigned by vmid which is set on line 39. vmid on line is only set if user send arg[:vmid] to 0 or leave it nil. 

We need args[:vmid] to have some value to create vm in line 56.

to reduce the AbcSize, we need to set both vmid and args[:vmid] if they are < 1. Even if we call different function. still there will be one/two assignments depending on condition (it's not necessary args[:vmid] is nil but still vmid < 1)

Comment (sbernhard) [reply app/models/foreman_fog_proxmox/proxmox_vm_commands.rb:40]: Thanks for the explanation. Can we somehow improve this code to make it more readable. You know the magic behind but nobody would understand it from the written code.

**Review by sbernhard — APPROVED** [review]
LGTM. Thanks.

**Comment by sbernhard** [comment]
would be good to fix the tests again.

**Pull request merged**